pulsar-broker/src/test/java/org/apache/pulsar/client/api/TopicReaderTest.java

```diff
@@ -650,4 +647,44 @@ public void testReaderIsAbleToSeekWithTimeOnMiddleOfTopic() throws Exception {
         reader.close();
         producer.close();
     }
+
+    @Test
+    public void testReaderStartMessageIdInclusiveCurrentPos() throws Exception {
+        final int numOfMessage = 10;
+        final String topicName = "persistent://my-property/my-ns/testReaderStartMessageIdInclusiveCurrentPos";
+        final int resetIndex = new Random().nextInt(numOfMessage); // Choose some random index to reset
+
+        Producer<byte[]> producer = pulsarClient.newProducer()
```

Comment:
Seems default producer is batch enabled, How about also add a test for batch disable? @lovelle

Reply:
We could have the test parametrized, since every other aspect should be the same

Reply:
Done, sorry for the delay.
Please take a look now at the parameters covered, If you can think of any additional just let me know and I will add it :+1: